CIP-DappConnector/CIP-dapp-connector.md

```diff
@@ -0,0 +1,226 @@
+---
+CIP: ?
+Title: Cardano dApp-Wallet Web Bridge
+Authors: rooooooooob
+Status: Draft
+Type: Standards
+Created: 2021-04-29
+License: CC-BY-4.0
+---
+
+# Abstract
+
+This documents describes a webpage-based communication bridge allowing webpages (i.e. dApps) to interface with Cardano wallets. This is done via injected javascript code into webpages. This specification defines the manner that such code is to be accessed by the webpage/dApp, as well as defining the API for dApps to communicate with the user's wallet. This document currently concerns the Shelley-Mary era but will have a second version once Plutus is supported. This specification is intended to cover similar use cases as web3 for Ethereum or [EIP-0012](https://github.com/ergoplatform/eips/pull/23) for Ergo. The design of this spec was based on the latter.
```

Comment:
Do we mean here by "Cardano wallets" actually Cardano accounts as per CIP-1852, correct? Some "Cardano wallets" (in the sense of a wallet app), like Yoroi have multiple logical cardano wallets (from different mnemonics/hw wallets), and Adalite even allows multiple accounts (using the CIP-1852 account index) for the same logical wallet

Reply:
For background on the wording: in Yoroi a wallet is just a public key (usually the account public key, but not necessarily). Everything in this spec was written with this as the definition of "wallet" in mind (for example, when you're granting read access, Yoroi just lists all the public keys in the wallet). This was done on purpose to allow for things like multi-account and standards other than CIP1852 in the future.